### PR TITLE
71: Support JUnit5

### PR DESCRIPTION
**Pull request by thodorisbais**
Open this PR as WIP so that we gradually move to JUnit 5.

---
This change is [Reviewable](https://reviewable.io/reviews/unitsofmeasurement/unit-api/152)

### CODE REVIEW

**Comment by keilw** [comment]
@thodorisbais This one caused a conflict with some of the other PRs, could you please resolve it or raise it again?

**Comment by thodorisbais** [comment]
Oh, yes, #148 was retrieved from my local stash in order to have a successful build before submitting this PR. I will revert #148 from this PR and have an overall look again at it.
Thus, I suspect it currently fails because it contains an already merged change (#148), but I am currently not at my computer, so, I'll let you know tomorrow.

**Comment by keilw** [comment]
No it does not seem because of the code, but some obscure Bintray problem. And they have no message creation or support for Open Source accounts, so I'll have to ask on Twitter again what is wrong.

**Comment by thodorisbais** [comment]
Please ignore my previous comments. I think I now got it.

Long story short, there are a few points we need to keep in mind:

- JUnit5 is indeed the way forward and will enable us to have better unit testing; thus, we need to indeed migrate to JUnit5
- Given that we already started with JUnit4, the best call for now is to just prepare the ground for JUnit 5; that is, we ultimately needed a JUnit5-ready setup ( JUnit5-ready = an environment which is able to run JUnit4 tests, but also supports JUnit5). The latter is herewith now possible with the `junit-vintage-engine`. So, now, with this PR we just updated our project to JUnit5, but nothing broke, since we are able to run JUnit4 tests on the JUnit5 platform
- From now on, we can start writing JUnit5 tests (this is the role of the rest of added dependencies and the reason behind removing the JUnit4 dependency)
- It's always our call to decide what to do with the current-written-with-JUnit4-engine JUnit tests; my take is to migrate those ones to JUnit5 as well (so that we are consistent), but given that it's a big change, let's start small by just supporting it in our project and we can then start migrating/rewriting smaller chunks

What do you think @keilw , @Daniel-Dos ?
Once this is merged, I can registed a new issue for migrating the currently existing JUnit4 tests to JUnit5 and assign it to myself; I'd love to work on it - JUnit5 looks cool.

File: pom.xml [hunk 1]
```diff
@@ -79,6 +79,8 @@
         <maven.jar.version>3.1.0</maven.jar.version>
         <maven.javadoc.version>3.0.1</maven.javadoc.version>
         <jacoco.plugin.version>0.8.2</jacoco.plugin.version>
+        <junit.jupiter.version>5.1.0</junit.jupiter.version>
```
Comment (otaviojava) [inline pom.xml]: why not the lastest version?
https://mvnrepository.com/artifact/org.junit.jupiter/junit-jupiter-engine/5.3.2

Comment (thodorisbais) [reply pom.xml]: That's right, @otaviojava , I probably overlooked this while pasting the dependencies from somewhere else. Updated both dependencies to their latest version.

**Comment by thodorisbais** [comment]
And yes! The build is green again! Was it random @keilw or you had to indeed get in touch with twitter?

**Comment by keilw** [comment]
No it wasn't random (at least not from our side) but thanks to @jbaruch from JFrog/Bintray they got it to work again.

**Comment by thodorisbais** [comment]
Cool, big thanks to @jbaruch then! 🎉

**Pull request merged**